src/commands/sns/follow_neuron.rs

```diff
@@ -0,0 +1,93 @@
+use candid::Encode;
+use clap::{ArgAction, ArgGroup, Parser, ValueEnum};
+use ic_sns_governance::pb::v1::{
+    manage_neuron::{Command, Follow},
+    ManageNeuron, NeuronId,
+};
+
+use crate::lib::{
+    signing::{sign_ingress_with_request_status_query, IngressWithRequestId},
+    AnyhowResult, AuthInfo, ROLE_SNS_GOVERNANCE,
+};
+
+use super::{ParsedSnsNeuron, SnsCanisterIds};
+
+/// Configures a neuron to follow another neuron or group of neurons. Following neurons
+/// causes your neuron to automatically vote when the majority (by voting power) of the
+/// followed neurons do.
+///
+/// Follow relationships are granular by the purpose of a proposal, e.g. you can follow a neuron
+/// only for votes on transferring SNS treasury funds. There are several built-in proposal
+/// functions which this command accepts by name, but new ones can be added at any time via
+/// AddGenericNervousSystemFunction proposals and those must be addressed by integer ID.
+#[derive(Parser)]
+#[clap(
+    group(ArgGroup::new("function").required(true)),
+    group(ArgGroup::new("following").required(true)),
+)]
+pub struct FollowNeuronOpts {
+    /// The neuron to configure.
+    neuron_id: ParsedSnsNeuron,
+    /// The name of the built-in proposal function to restrict following to.
+    #[clap(long, group = "function", value_enum)]
+    r#type: Option<Type>,
+    /// The numeric ID of the proposal function to restrict following to.
+    #[clap(long, group = "function")]
+    function_id: Option<u64>,
+    /// A list of neurons to follow for this proposal function, separated by commas.
+    #[clap(long, action = ArgAction::Append, value_delimiter = ',', group = "following")]
+    followees: Vec<ParsedSnsNeuron>,
+    /// Remove any followees for this proposal function.
+    #[clap(long, group = "following")]
+    unfollow: bool,
```

Comment:
When first reading the command-line options (though not too closely), I expected that `--unfollow --followees <someone>` would unfollow a single followee, but leave the rest in place (for that function).

It might be good to clarify in the help that the specified followees replace any previous followees for the function.

Two suggestions:
1. rename `--unfollow` to `--unfollow-all`
2. make sure it's a command-line error to try to pass both `--unfollow-all` and `--followees`

```suggestion
    /// A list of neurons to follow for this proposal function, separated by commas.
    #[clap(long, action = ArgAction::Append, value_delimiter = ',', group = "following", conflicts_with="unfollow_all")]
    followees: Vec<ParsedSnsNeuron>,
    /// Remove any followees for this proposal function.
    #[clap(long, group = "following", conflicts_with="followees")]
    unfollow_all: bool,
```

Reply:
That's what ArgGroup does. The help text makes it explicit, `<--followees <FOLLOWEES>|--unfollow>`